CRM/Core/Payment.php

```diff
@@ -150,32 +142,37 @@ abstract class CRM_Core_Payment {
   protected $backOffice = FALSE;
 
   /**
-   * @return bool
-   */
-  public function isBackOffice() {
-    return $this->backOffice;
-  }
-
-  /**
-   * Set back office property.
+   * This is only needed during the transitional phase. In future you should
+   * pass your own PropertyBag into the method you're calling.
    *
-   * @param bool $isBackOffice
+   * New code should NOT use $this->propertyBag.
+   *
+   * @var Civi\Payment\PropertyBag
    */
-  public function setBackOffice($isBackOffice) {
-    $this->backOffice = $isBackOffice;
-  }
+  protected $propertyBag;
 
   /**
-   * Get payment instrument id.
+   * Returns an explicitly-set paymentInstrumentID but default to the
+   * payment processor's configured one if it's not been set.
+   *
+   * This is akward in relation to the new propertyBag. Recommend that future
+   * payment processor code does the logic below on it's own passed-in propertyBag
+   * and does not use this method.
    *
    * @return int
    */
   public function getPaymentInstrumentID() {
-    return $this->paymentInstrumentID ? $this->paymentInstrumentID : $this->_paymentProcessor['payment_instrument_id'];
+    $propertyBag = $this->getInternalPropertyBag();
```

Comment:
The more I think about it the more this bothers me. The rule is that the payment instrument id is set in the civicrm_payment_processor table & it's a feature of the processor. I understand there is a suggestion some processors want to break that design principle but I think that needs to be it's own conversation. Without this change the current design rule stands

Reply:
yep!

Reply:
Yeah that makes sense - can we leave it in this PR (but remove from propertyBag) and follow up with a fix on that?